bundles/org.openhab.binding.senseenergy/src/main/java/org/openhab/binding/senseenergy/internal/handler/SenseEnergyProxyDeviceHandler.java

```diff
@@ -0,0 +1,334 @@
+import static org.openhab.binding.senseenergy.internal.SenseEnergyBindingConstants.CHANNEL_PROXY_DEVICE_SWITCH;
+import static org.openhab.binding.senseenergy.internal.SenseEnergyBindingConstants.CONFIG_PARAMETER_MAC;
+import static org.openhab.binding.senseenergy.internal.SenseEnergyBindingConstants.CONFIG_PARAMETER_POWER_LEVELS;
+
+import java.util.Arrays;
+import java.util.HexFormat;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Random;
+
+import javax.measure.quantity.Power;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.senseenergy.internal.api.dto.SenseEnergyDatagramGetRealtime;
+import org.openhab.binding.senseenergy.internal.api.dto.SenseEnergyDatagramGetSysInfo;
+import org.openhab.binding.senseenergy.internal.config.SenseEnergyProxyDeviceConfiguration;
+import org.openhab.binding.senseenergy.internal.handler.helpers.SenseEnergyPowerLevels;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @link { SenseEnergyProxyDeviceHandler }
+ *
+ * @author Jeff James - Initial contribution
+ */
+@NonNullByDefault
+public class SenseEnergyProxyDeviceHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(SenseEnergyProxyDeviceHandler.class);
+
+    private static final byte[] OUI = new byte[] { 0x53, 0x75, 0x31 };
+    private static final String PROXY_DEVICE_SW_VERSION = "1.2.5 Build 171206 Rel.085954";
+    private static final String PROXY_DEVICE_HW_VERSION = "1.0";
+    private static final String PROXY_DEVICE_TYPE = "IOT.SMARTPLUGSWITCH";
+    private static final String PROXY_DEVICE_MODEL = "HS110(US)";
+
+    private SenseEnergyProxyDeviceConfiguration config = new SenseEnergyProxyDeviceConfiguration();
+
+    SenseEnergyPowerLevels powerLevels = new SenseEnergyPowerLevels();
+
+    private ElectricalData electricalData = new ElectricalData();
+    private boolean selfConfigurationChange = false;
+
+    public SenseEnergyProxyDeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SenseEnergyProxyDeviceConfiguration.class);
+
```

Comment:
A configuration validation might be usefull here.

Reply:
could you elaborate on the request here?  i "believe" i am doing that validation in the intialize by looking for isBlank on powerLevels.  The other fields should be validated by the default validateConfigurationParameters in the BaseThingHandler, correct?

Reply:
Yes, not sure how i missed this or what happened, but the isBlank checks are exactly what i meant.